compiler_opt/rl/env.py

```diff
@@ -23,38 +25,31 @@
 import io
 import os
 import tempfile
-from typing import Any, Generator, List, Optional, Tuple, Type
+from typing import Callable, Generator, List, Optional, Tuple, Type
 
 import numpy as np
 
 from compiler_opt.rl import corpus
 from compiler_opt.rl import log_reader
 
-OBS_T = Any
-
-OBS_KEY = 'obs'
-REWARD_KEY = 'reward'
-SCORE_POLICY_KEY = 'score_policy'
-SCORE_DEFAULT_KEY = 'score_default'
-CONTEXT_KEY = 'context'
-MODULE_NAME_KEY = 'module_name'
-OBS_ID_KEY = 'obs_id'
-STEP_TYPE_KEY = 'step_type'
-
-FIRST_STEP_STR = 'first'
-MID_STEP_STR = 'mid'
-LAST_STEP_STR = 'last'
-
-_TERMINAL_OBS = {
-    OBS_KEY: {},
-    REWARD_KEY: 0.0,
-    SCORE_POLICY_KEY: 0.0,
-    SCORE_DEFAULT_KEY: 0.0,
-    CONTEXT_KEY: '',
-    MODULE_NAME_KEY: '',
-    OBS_ID_KEY: -1,
-    STEP_TYPE_KEY: LAST_STEP_STR,
-}
+
+class StepType(Enum):
+  FIRST = 1
+  MID = 2
+  LAST = 3
+
+
+@dataclasses.dataclass
+class TimeStep:
+  obs: Optional[dict[str, np.NDArray]]
+  reward: Optional[dict[str, float]]
+  score_policy: Optional[dict[str, float]]
```

Comment:
I think the standard use case that was expected was that of having a policy which we want to evaluate and a default (comparator) policy. The reward would then be the relative improvement of the policy to the comparator, e.g., as in the regalloc case. Currently I think the environment will always open two compiler processes, one for policy and one for the comparator, so score_policy and score_default are always computed anyway, and at that point we might as well compute the relative improvement as reward too.

Reply:
I think @jacob-hegna had it this way "if the user wants that data"; orthogonally, I want to be able to pass the baseline ("default") to avoid re-computing it.